src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/DiagnosticItem.cs

```diff
@@ -2,91 +2,105 @@
+    private static ImageMoniker MapEffectiveSeverityToIconMoniker(ReportDiagnostic effectiveSeverity)
+        => effectiveSeverity switch
         {
-            if (EffectiveSeverity != newEffectiveSeverity)
-            {
-                EffectiveSeverity = newEffectiveSeverity;
+            ReportDiagnostic.Error => KnownMonikers.CodeErrorRule,
+            ReportDiagnostic.Warn => KnownMonikers.CodeWarningRule,
+            ReportDiagnostic.Info => KnownMonikers.CodeInformationRule,
+            ReportDiagnostic.Hidden => KnownMonikers.CodeHiddenRule,
+            ReportDiagnostic.Suppress => KnownMonikers.CodeSuppressedRule,
+            _ => default,
+        };
+
+    private void NotifyPropertyChanged(string propertyName)
+    {
+        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+    }
 
-                NotifyPropertyChanged(nameof(EffectiveSeverity));
-                NotifyPropertyChanged(nameof(IconMoniker));
-            }
-        }
+    internal void SetRuleSetSeverity(ReportDiagnostic value, string pathToRuleSet)
+    {
+        var ruleSetDocument = XDocument.Load(pathToRuleSet);
 
-        private static ImageMoniker MapEffectiveSeverityToIconMoniker(ReportDiagnostic effectiveSeverity)
-            => effectiveSeverity switch
-            {
-                ReportDiagnostic.Error => KnownMonikers.CodeErrorRule,
-                ReportDiagnostic.Warn => KnownMonikers.CodeWarningRule,
-                ReportDiagnostic.Info => KnownMonikers.CodeInformationRule,
-                ReportDiagnostic.Hidden => KnownMonikers.CodeHiddenRule,
-                ReportDiagnostic.Suppress => KnownMonikers.CodeSuppressedRule,
-                _ => default,
-            };
-
-        private void NotifyPropertyChanged(string propertyName)
-        {
-            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
-        }
+        ruleSetDocument.SetSeverity(_analyzerReference.Display, Descriptor.Id, value);
 
-        internal void SetRuleSetSeverity(ReportDiagnostic value, string pathToRuleSet)
-        {
-            var ruleSetDocument = XDocument.Load(pathToRuleSet);
+        ruleSetDocument.Save(pathToRuleSet);
+    }
 
-            ruleSetDocument.SetSeverity(_analyzerReference.Display, Descriptor.Id, value);
+    internal Task<Solution> GetSolutionWithUpdatedAnalyzerConfigSeverityAsync(ReportDiagnostic value, Project project, CancellationToken cancellationToken)
+    {
+        var effectiveSeverity = value.ToDiagnosticSeverity() ?? Descriptor.DefaultSeverity;
+        var diagnostic = Diagnostic.Create(Descriptor, Location.None, effectiveSeverity, additionalLocations: null, properties: null);
+        return ConfigurationUpdater.ConfigureSeverityAsync(value, diagnostic, project, cancellationToken);
+    }
 
-            ruleSetDocument.Save(pathToRuleSet);
-        }
+    public override int GetHashCode()
```

Comment:
equals/gethashcode are used in next PR as part of proper items in a observable-collectoin and virtualizing-tree-view.